src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java

```diff
@@ -0,0 +1,347 @@
+            }
+        } else {
+            LOGGER.debug("Starting inserts in http thread");
+            res = callables.get(0).call();
+        }
+        long c = System.nanoTime();
+        LOGGER.debug("read {} took {}ms (count {}ms, {} queries in {} threads {}ms)", uuid, (c - a) / 1000000,
+                (b - a) / 1000000, cnt, threadcount, (c - b) / 1000000);
+
+        // TODO same as save, avoid the transpose to allow stream from database to
+        // clients ?
+        // TODO avoid this extra copy to an intermediate transposed map
+        Map<String, List<Object>> data = new HashMap<>();
+        for (Map.Entry<Object, Object> entry : res.entrySet()) {
+            Map<Object, Object> dict = (Map<Object, Object>) entry.getValue();
+            for (Map.Entry<Object, Object> entryPoint : dict.entrySet()) {
+                String tsname = (String) entryPoint.getKey();
+                // TODO more types
+                Object val = entryPoint.getValue();
+                data.computeIfAbsent(tsname, _ignored -> new ArrayList<>()).add(val);
+            }
+        }
+        List<TimeSeries> ret = new ArrayList<>();
+        for (Map.Entry<String, List<Object>> entry : data.entrySet()) {
+            // TODO remove duplication
+            if (entry.getValue().get(0) instanceof Double) {
+                double[] doubles = entry.getValue().stream().map(Double.class::cast).mapToDouble(Double::doubleValue)
+                        .toArray();
+
+                // TODO should be in the timeseries API ?
+                DoubleDataChunk ddc = new UncompressedDoubleDataChunk(0, doubles);
+                // TODO get compress mode from the metadata sent by the client
+                if (tryToCompress) {
+                    ddc = ddc.tryToCompress();
+                }
+                // TODO more types
+                // TODO index from client
+                TimeSeries timeseries = new StoredDoubleTimeSeries(timeseriesMetadataService.getMetadata(index, individualMetadatas, entry.getKey()), List.of(ddc));
+                ret.add(timeseries);
+            } else if (entry.getValue().get(0) instanceof String) {
+                String[] strings = entry.getValue().toArray(new String[0]);
+
+                // TODO should be in the timeseries API ?
+                StringDataChunk ddc = new UncompressedStringDataChunk(0, strings);
+                // TODO get compress mode from the metadata sent by the client
+                if (tryToCompress) {
+                    ddc = ddc.tryToCompress();
+                }
+                // TODO more types
+                // TODO index from client
+                TimeSeries timeseries = new StringTimeSeries(timeseriesMetadataService.getMetadata(index, individualMetadatas, entry.getKey()), List.of(ddc));
+                ret.add(timeseries);
+            } else {
+                throw new RuntimeException("Unsupported read of timeseries type" + entry.getValue().get(0).getClass());
```

Comment:
As above, in my opinion, checking type by 'dataType' is a better choice. I try to test with rest endpoint to get an empty time line, an exception 
java.lang.NullPointerException: null
	at org.gridsuite.timeseries.server.TimeseriesDataRepository.dofindById(TimeseriesDataRepository.java:324) ~[classes/:na]
	at org.gridsuite.timeseries.server.TimeseriesDataRepository.findById(TimeseriesDataRepository.java:200) ~[classes/:na]
	at org.gridsuite.timeseries.server.TimeseriesDataRepository$$FastClassBySpringCGLIB$$1ac395ff.invoke(<generated>) ~[classes/:na]  

An example of empty timeline produce by dynawalz

"timeLine" : {
    "metadata" : {
      "name" : "timeLine",
      "dataType" : "STRING",
      "tags" : [ ],
      "regularIndex" : {
        "startTime" : 0,
        "endTime" : 0,
        "spacing" : 0
      }
    },
    "chunks" : [ ]
  }